osgtest/tests/test_155_stashcache.py

```diff
@@ -0,0 +1,157 @@
+xrd.allow host *
+sec.protocol  host
+all.adminpath /var/spool/xrootd
+
+xrootd.trace emsg login stall redirect
+ofs.trace all
+xrd.trace all
+cms.trace all
+
+ofs.osslib  libXrdPss.so
+# normally this is the redirector but we don't have one in this environment
+pss.origin localhost:{ORIGIN_XROOT_PORT}
+pss.cachelib libXrdFileCache.so
+pss.setopt DebugLevel 1
+
+oss.localroot $(cachedir)
+
+pfc.blocksize 512k
+pfc.ram       1024m
+# ^ xrootd won't start without a gig
+pfc.prefetch  10
+pfc.diskusage 0.90 0.95
+
+ofs.authorize 1
+acc.audit deny grant
+
+acc.authdb {CACHE_AUTHFILE_PATH}
+sec.protbind * none
+xrd.protocol http:{CACHE_HTTP_PORT} libXrdHttp.so
+
+xrd.port {CACHE_XROOT_PORT}
+
+http.listingdeny yes
+http.staticpreload http://static/robots.txt /etc/xrootd/stashcache-robots.txt
+
+# Tune the client timeouts to more aggressively timeout.
+pss.setopt ParallelEvtLoop 10
+pss.setopt RequestTimeout 25
+#pss.setopt TimeoutResolution 1
+pss.setopt ConnectTimeout 25
+pss.setopt ConnectionRetry 2
+#pss.setopt StreamTimeout 35
+
+all.sitename osgtest
+
+xrootd.diglib * /etc/xrootd/digauth.cf
+""".format(**globals())
+
+
+CACHE_AUTHFILE_TEXT = """\
+u * / rl
+"""
+
+
+ORIGIN_CONFIG_TEXT = """\
+xrd.allow host *
+sec.protocol  host
+sec.protbind  * none
+all.adminpath /var/spool/xrootd
+all.pidpath /var/run/xrootd
+
+# The directory on local disk containing the files to share, e.g. "/stash".
+oss.localroot {ORIGIN_DIR}
+all.export /
+
+xrd.port {ORIGIN_XROOT_PORT}
+all.role server
+
+xrootd.trace emsg login stall redirect
+ofs.trace all
+xrd.trace all
+cms.trace all
+""".format(**globals())
+
+
+CACHES_JSON_TEXT = """\
+[
+{"name":"root://localhost", "status":1, "longitude":-89.4012, "latitude":43.0731}
+]
+"""
+
+
+_NAMESPACE = "stashcache"
+
+
+def _getcfg(key):
+    return core.config["%s.%s" % (_NAMESPACE, key)]
+
+
+def _setcfg(key, val):
+    core.config["%s.%s" % (_NAMESPACE, key)] = val
+
+
+class TestStartStashCache(OSGTestCase):
+    @core.elrelease(7,8)
+    def setUp(self):
+        core.skip_ok_unless_installed("stashcache-origin-server", "stashcache-cache-server", "stashcache-client")
+
+    def test_01_configure(self):
+        for key, val in [
+            ("cache_authfile_path", CACHE_AUTHFILE_PATH),
+            ("cache_config_path", CACHE_CONFIG_PATH),
+            ("origin_config_path", ORIGIN_CONFIG_PATH),
+            ("caches_json_path", CACHES_JSON_PATH),
+            ("cache_http_port", CACHE_HTTP_PORT),
+            ("origin_dir", ORIGIN_DIR),
+            ("cache_dir", CACHE_DIR),
+            ("origin_xroot_port", ORIGIN_XROOT_PORT),
+            ("cache_xroot_port", CACHE_XROOT_PORT)
+        ]:
+            _setcfg(key, val)
+
+        xrootd_user = pwd.getpwnam("xrootd")
+        for d in [_getcfg("origin_dir"), _getcfg("cache_dir"),
+                  os.path.dirname(_getcfg("caches_json_path"))]:
+            files.safe_makedirs(d)
+            os.chown(d, xrootd_user.pw_uid, xrootd_user.pw_gid)
```

Comment:
What do you think about making `files.safe_makedirs` take optional args for chown/chmod (defaulting to `root:root` and `0644`)?

Reply:
I'd rather have it follow the semantics of its namesake (`os.makedirs`) as opposed to making something different and unexpected.

But you just reminded me that the chown is redundant anyway since osg-test always runs as root.

Reply:
Maybe I'm misunderstanding your last comment but the chown that you removed is setting ownership to `xrootd:xrootd` and now it's chowning the dirs to `root:root`

Reply:
/me looks at code again
I am a moron. Fixed.